packages/nuxt/src/pages/runtime/page.ts

```diff
@@ -88,6 +89,10 @@ export default defineComponent({
           }
 
           const key = generateRouteKey(routeProps, props.pageKey)
+          if (!nuxtApp.isHydrating && previousPageKey === key) {
```

Comment:
should we use `isChangingPage`?

Reply:
I don't know 🤔 , `isChangingPage` is relying on `to` and `from` from a router guard. 

This implementation is relying on whether we force rerender the page or not by comparing the generated key.

Do you think we should use a router guard instead ?

Reply:
whats the solution

Reply:
Do we want to have a different behavior compared to `isChangingPage` for checking? If not, let's use it.

Reply:
I think we want to stop the loading indicator only if the setup of the new page won't be run. 

Using a router guard with beforeResolve don't necessaraly means that the setup will be re-run 🤔.